### PR TITLE
Fix delivery service validation

### LINKED ISSUE #2373: Missing validation in Golang Delivery Service API
Delivery Service API validation that was done in the Perl API is missing from the current Golang API:

1. `typeId` is a valid Delivery Service type [(see here)](https://github.com/apache/incubator-trafficcontrol/blob/master/traffic_ops/app/lib/API/Deliveryservice.pm#L1537)
2. `missLat` is a valid latitude [(see here)](https://github.com/apache/incubator-trafficcontrol/blob/master/traffic_ops/app/lib/API/Deliveryservice.pm#L1548)
3. `missLong` is a valid longitude [(see here)](https://github.com/apache/incubator-trafficcontrol/blob/master/traffic_ops/app/lib/API/Deliveryservice.pm#L1572)
4. `initialDispersion` must be 1 or greater [(see here)](https://github.com/apache/incubator-trafficcontrol/blob/master/traffic_ops/app/lib/API/Deliveryservice.pm#L1562)
5. `orgServerFqdn` has a valid `<protocol>://<fqdn>[:<port>]` format [(see here)](https://github.com/apache/incubator-trafficcontrol/blob/master/traffic_ops/app/lib/API/Deliveryservice.pm#L1586)

### PR DESCRIPTION
**Pull request by rawlinp**
Add the rest of the validation done by the Perl API that the Go API is
missing.

Fixes #2373

### CODE REVIEW

File: lib/go-tc/deliveryservices.go [hunk 1]
```diff
@@ -241,15 +241,20 @@ func (ds *DeliveryServiceNullableV12) Sanitize() {
 	}
 }
 
-func getTypeName(tx *sql.Tx, id int) (string, bool, error) {
+func getTypeData(tx *sql.Tx, id int) (string, string, bool, error) {
```
Comment (rob05c) [inline lib/go-tc/deliveryservices.go]: Mind adding a Godoc function comment here, explaining the return params? That way someone doesn't have to read the function to figure out what each string being returned is.

File: lib/go-tc/tovalidate/rules.go [hunk 2]
```diff
@@ -46,3 +48,19 @@ func IsOneOfStringICase(set ...string) func(string) bool {
 	}
 	return IsOneOfString(lowcased...)
 }
+
+func IsGreaterThanZero(value interface{}) error {
```
Comment (dewrich) [inline lib/go-tc/tovalidate/rules.go]: This function could be refactored to here: https://github.com/apache/trafficcontrol/blob/master/lib/go-tc/tovalidate/rules.go#L19

File: lib/go-tc/deliveryservices.go (hunk 1 above)
Comment (rawlinp) [reply lib/go-tc/deliveryservices.go]: done

File: lib/go-tc/tovalidate/rules.go (hunk 2 above)
Comment (rawlinp) [reply lib/go-tc/tovalidate/rules.go]: am confused, this function _is_ there

**Comment by rawlinp** [comment]
hold up, this makes `initialDispersion` required...fixing

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1882/
Test PASSed.

**Comment by rawlinp** [comment]
ok, fixed it so `initialDispersion` has to be greater than zero if it's not nil.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1884/
Test PASSed.

File: lib/go-tc/tovalidate/rules.go (hunk 2 above)
Comment (dewrich) [reply lib/go-tc/tovalidate/rules.go]: sorry I didn't notice the package changed from above, ignore my previous comment

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1909/
Test PASSed.

**Pull request merged**